Version 1.65.0

WilliamDEdwards opened the pull request:
# Changes

Add CMS options endpoint.

The `name` path parameter is an Enum. Other endpoints that need an Enum do not use the `valueIn` validation that models use for Enums. I don't know if there is a reason for that. If there is, commit https://github.com/vdhicts/cyberfusion-cluster-api-client/commit/7e1476ca8a816dca450cb39cb9ebecba5b51909d can be easily dropped.

# Checks

- [x] The version constant is updated in `Client.php`
- [x] The changelog is updated (when applicable)
- [x] The supported Cluster API version is updated in the README (when applicable)

dvdheiden commented:
About the validation: I tend to validate the models and then provide the models to the endpoints so we don't have to put validation logic in the endpoints (except that the, for that endpoint required fields, are filled). So in this case, that would be:

```php
public function updateOption(int $id, CmsOption $cmsOption): Response
```

That would mean we need to add the name to the model, which might make sense so you always know which option the value belongs to.

WilliamDEdwards commented:
> About the validation: I tend to validate the models and then provide the models to the endpoints so we don't have to put validation logic in the endpoints (except that the, for that endpoint required fields, are filled). So in this case, that would be:
> 
> ```
> public function updateOption(int $id, CmsOption $cmsOption): Response
> ```
> 
> That would mean we need to add the name to the model, which might make sense so you always know which option the value belongs to.

In the Cluster API, `name` is not yet part of the `CMSOption` model. It is specified only as a path parameter. 

Commit 2923715a9c7010e6b06ba7ab5be4a596011af3cd adds `name` to the model in this library. It tries to set `name` from the response body. The Cluster API must be updated to return this attribute. I will update this issue once that has been done.

WilliamDEdwards commented:
> > About the validation: I tend to validate the models and then provide the models to the endpoints so we don't have to put validation logic in the endpoints (except that the, for that endpoint required fields, are filled). So in this case, that would be:
> > ```
> > public function updateOption(int $id, CmsOption $cmsOption): Response
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > That would mean we need to add the name to the model, which might make sense so you always know which option the value belongs to.
> 
> In the Cluster API, `name` is not yet part of the `CMSOption` model. It is specified only as a path parameter.
> 
> Commit [2923715](https://github.com/vdhicts/cyberfusion-cluster-api-client/commit/2923715a9c7010e6b06ba7ab5be4a596011af3cd) adds `name` to the model in this library. It tries to set `name` from the response body. The Cluster API must be updated to return this attribute. I will update this issue once that has been done.

The Cluster API has been updated. The `name` property has been added to the response body. The request payload has not been changed. As far as I'm concerned, this PR can now be merged.

dvdheiden commented:
Thank you, I will take a look at this later today.

WilliamDEdwards commented:
To prevent a proliferation of versions, I integrated the changes of https://github.com/vdhicts/cyberfusion-cluster-api-client/pull/135 and https://github.com/vdhicts/cyberfusion-cluster-api-client/pull/136 into this branch:

- I cherry-picked dc38b34e4f427bee5e6ed6edcf5a93f1a8698dec and 31f63068b67fa286282b9f596aff22fe50d9b4aa from https://github.com/vdhicts/cyberfusion-cluster-api-client/pull/135, and fixed up d55c87ea9546227ecce907d52d6204deecbcf924 into 31f63068b67fa286282b9f596aff22fe50d9b4aa.
- I cherry-picked d908e2dcef5e456539baca1bbf59d193f93bb897 from https://github.com/vdhicts/cyberfusion-cluster-api-client/pull/136, and fixed up fb0ddee1c1b4df494f60afb14b7659bf5b71106a into d908e2dcef5e456539baca1bbf59d193f93bb897.
- I merged the 1.65.0 changelog with the 1.66.0 and 1.67.0 changelogs
- I closed the PRs for 1.66.0 and 1.67.0
